protocol/communities/manager.go

```diff
@@ -4720,6 +4748,19 @@ func (m *Manager) handleCommunityTokensMetadata(community *Community) error {
 	return nil
 }
 
+func (m *Manager) handleCommunityGrant(communityID types.HexBytes, grant []byte, clock uint64) error {
+	_, oldClock, err := m.persistence.GetCommunityGrant(communityID.String())
+	if err != nil {
+		return err
+	}
+
+	if oldClock >= clock {
+		return nil
+	}
+
+	return m.persistence.SaveCommunityGrant(communityID.String(), grant, clock)
```

Comment:
Grant comes from the control node in the case it should be handled by `handleCommunityGrant`. Not sure it it can be invalid. WDYT @osmaczko?

Reply:
Currently this function is invoked only from `HandleCommunityRequestToJoinResponse`. The response to join is validated against the control node, and as a consequence, we don't need to double-check it here. However, it would be more correct to validate the signature here as well, because this function may be used in different contexts soon.

Reply:
Added it to list item here: https://github.com/status-im/status-go/issues/4724